packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -266,12 +275,21 @@ const BlockInspectorSingleBlock = ( {
 						<BlockStylesPanel clientId={ clientId } />
 					) }
 
-					{ contentClientIds && contentClientIds?.length > 0 && (
-						<PanelBody title={ __( 'Content' ) }>
-							<BlockQuickNavigation
-								clientIds={ contentClientIds }
-							/>
-						</PanelBody>
+					{ ! isNavigationBlock &&
```

Comment:
We could make this condition some kind of private editor setting. We can then pass in the various blocks we want to enable this behaviour for.